### PR TITLE
NODE-1964: UpdateAssetInfoTransaction tests

### PR DESCRIPTION
**Pull request by finnoio**

### CODE REVIEW

File: node-it/src/test/scala/com/wavesplatform/it/sync/activation/VRFProtobufActivationSuite.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  test("not able to broadcast UpdateAssetInfoTransaction before activation") {
+    sender.waitForHeight(sender.transactionInfo(assetId).height + updateInterval + 1)
+    assertApiError(sender.updateAssetInfo(issuer, assetId, "updatedName", "updatedDescription", minFee)) { error =>
+      error.statusCode shouldBe 400
+      error.message shouldBe "State check failed. Reason: ActivationError(VRF and Protobuf feature has not been activated yet)"
+      error.id shouldBe 112
+    }
+  }
+
+  test("able to broadcast UpdateAssetInfoTransaction after activation") {
```
Comment (msmolyakov) [inline node-it/src/test/scala/com/wavesplatform/it/sync/activation/VRFProtobufActivationSuite.scala]: Is it possible to rename a token at the height of activation, if it is issued before activation and `updateInteral` has not yet been reached?

File: node-it/src/test/scala/com/wavesplatform/it/sync/grpc/UpdateAssetInfoTransactionGrpcSuite.scala [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+import scala.util.Random
+import scala.concurrent.duration._
+
+class UpdateAssetInfoTransactionGrpcSuite extends GrpcBaseTransactionSuite with TableDrivenPropertyChecks{
+  import UpdateAssetInfoTransactionGrpcSuite._
+  val updateInterval = 2
+  override protected def nodeConfigs: Seq[Config] = Seq(configWithUpdateIntervalSetting(updateInterval).withFallback(Miners.head))
+
+  val issuer = KeyPair("issuer".getBytes)
+  val nonIssuer = KeyPair("nonIssuer".getBytes)
```
Comment (msmolyakov) [inline node-it/src/test/scala/com/wavesplatform/it/sync/grpc/UpdateAssetInfoTransactionGrpcSuite.scala]: Why not to use the predefined first/second addresses?

File: node-it/src/test/scala/com/wavesplatform/it/sync/grpc/UpdateAssetInfoTransactionGrpcSuite.scala [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      "Can't update asset info before",
+      Code.INVALID_ARGUMENT
+    )
+  }
+
+  val invalidAssetsNames =
+    Table(
+      "abc",
+      "UpperCaseAssetCoinTest",
+      "~!|#$%^&*()_+=\";:/?><|\\][{}"
```
Comment (msmolyakov) [inline node-it/src/test/scala/com/wavesplatform/it/sync/grpc/UpdateAssetInfoTransactionGrpcSuite.scala]: empty string and null? Longer name than limit?

File: node-it/src/test/scala/com/wavesplatform/it/sync/grpc/UpdateAssetInfoTransactionGrpcSuite.scala [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  test("not able to update info of not-issued asset") {
+    val notIssuedAssetId = "BzARFPgBqWFu6MHGxwkPVKmaYAzyShu495Ehsgru72Wz"
+    assertGrpcError(
+      sender.updateAssetInfo(issuer, notIssuedAssetId, "updatedName", "updatedDescription", minFee),
+      "Referenced assetId not found",
+      Code.INVALID_ARGUMENT
+    )
+  }
+
+  test("non-issuer cannot update asset info") {
```
Comment (msmolyakov) [inline node-it/src/test/scala/com/wavesplatform/it/sync/grpc/UpdateAssetInfoTransactionGrpcSuite.scala]: Is there unit tests for smart sender/token extra fees?

File: node-it/src/test/scala/com/wavesplatform/it/api/SyncHttpApi.scala [hunk 5]
```diff
@@ -9,8 +10,12 @@ import com.google.protobuf.ByteString
...
 import com.wavesplatform.account.{AddressOrAlias, KeyPair}
 import com.wavesplatform.api.grpc.BalanceResponse.WavesBalances
-import com.wavesplatform.api.grpc.{AccountRequest, AccountsApiGrpc, BalancesRequest, BlockRangeRequest, BlockRequest, BlocksApiGrpc, DataRequest, ScriptData, TransactionsApiGrpc, TransactionsRequest}
-import com.wavesplatform.api.grpc.{AccountRequest, AccountsApiGrpc, BalancesRequest, BlocksApiGrpc, DataRequest, ScriptData, TransactionsApiGrpc, TransactionsRequest}
+
+import com.wavesplatform.api.grpc._//{AccountsApiGrpc, AssetInfoResponse, BalancesRequest, BlockRangeRequest, BlockRequest, BlocksApiGrpc}
+
+//import com.wavesplatform.api.grpc.{AccountRequest, AccountsApiGrpc, BalancesRequest, BlockRangeRequest, BlockRequest, BlocksApiGrpc, DataRequest, ScriptData, TransactionsApiGrpc, TransactionsRequest}
+//import com.wavesplatform.api.grpc.{AccountRequest, AccountsApiGrpc, BalancesRequest, BlocksApiGrpc, DataRequest, ScriptData, TransactionsApiGrpc, TransactionsRequest}
+//>>>>>>> master
```
[the remark this answers is not in the document]
Comment (finnoio) [reply node-it/src/test/scala/com/wavesplatform/it/api/SyncHttpApi.scala]: done

File: node-it/src/test/scala/com/wavesplatform/it/sync/grpc/UpdateAssetInfoTransactionGrpcSuite.scala (hunk 2 above)
Comment (finnoio) [reply node-it/src/test/scala/com/wavesplatform/it/sync/grpc/UpdateAssetInfoTransactionGrpcSuite.scala]: fixed here https://github.com/wavesplatform/Waves/pull/2871/commits/471faf43564f012105b516053ceafbef113bbdcf

File: node-it/src/test/scala/com/wavesplatform/it/sync/activation/VRFProtobufActivationSuite.scala (hunk 1 above)
Comment (finnoio) [reply node-it/src/test/scala/com/wavesplatform/it/sync/activation/VRFProtobufActivationSuite.scala]: No, it is not possible. Here is the test -
https://github.com/wavesplatform/Waves/pull/2871/files#diff-2436f8954c013c29d97a70512759dfa0R43

File: node-it/src/test/scala/com/wavesplatform/it/sync/grpc/UpdateAssetInfoTransactionGrpcSuite.scala (hunk 4 above)
Comment (finnoio) [reply node-it/src/test/scala/com/wavesplatform/it/sync/grpc/UpdateAssetInfoTransactionGrpcSuite.scala]: There is no unit test for it, but here is an integration test
https://github.com/wavesplatform/Waves/pull/2871/files#diff-04882afc75a1259824fc515be37793dcR145

File: node-it/src/test/scala/com/wavesplatform/it/sync/grpc/UpdateAssetInfoTransactionGrpcSuite.scala (hunk 3 above)
Comment (finnoio) [reply node-it/src/test/scala/com/wavesplatform/it/sync/grpc/UpdateAssetInfoTransactionGrpcSuite.scala]: done, except null (which is impossible to pass)

**Pull request merged**